app/services/record_declarations/base.rb

```diff
@@ -88,7 +88,7 @@ def record_exists_with_different_declaration_date?
         declaration_type: declaration_type,
       )
 
-      declaration.present? && declaration.declaration_date != Time.zone.parse(declaration_date)
```

Comment:
I think I'd put the condition in the finder because it keeps the main condition of this method clear.

```ruby
      declaration = self.class.declaration_model.where.not(state: :voided).
        .find_by(
        user: user,
        course_identifier: course_identifier,
        declaration_type: declaration_type,
      )
```

etc.

Reply:
i'll correct this in this PR https://github.com/DFE-Digital/early-careers-framework/pull/1605/files this I'll have to apply the same fix to the duplicate queries